db/seeds.rb

```diff
@@ -1,3 +1,102 @@
+task_processing = Task.find_or_create_by!(
+  status: :in_progress,
+  due_on: due_date,
+  framework: framework_processing,
+  supplier: supplier,
+  period_month: Date.today.month,
+  period_year: Date.today.year,
+  description: 'In progress task (processing submission)'
+)
+processing_submission = supplier.submissions.find_or_create_by!(framework: framework_processing, task: task_processing, aasm_state: "processing")
+submission_file = processing_submission.files.find_or_create_by!(rows: 2)
+processing_submission.entries.find_or_create_by!(submission_file: submission_file, data: { key: "value" }, source: { sheet: "InvoicesReceived", row: 1 })
+processing_submission.entries.find_or_create_by!(submission_file: submission_file, data: { key: "another value" }, source: { sheet: "InvoicesReceived", row: 2 })
+
+
+framework_valid = Framework.find_or_create_by!(short_name: 'cboard11', name: 'Cheese Board 11')
+Agreement.find_or_create_by!(supplier: supplier, framework: framework_valid)
+task_in_review = Task.find_or_create_by!(
+  status: :in_progress,
+  due_on: due_date,
+  framework: framework_valid,
+  supplier: supplier,
+  period_month: Date.today.month,
+  period_year: Date.today.year,
+  description: 'In review task (validated submission)'
+)
+valid_submission = supplier.submissions.find_or_create_by!(framework: framework_valid, task: task_in_review, aasm_state: "in_review")
+submission_file = valid_submission.files.find_or_create_by!(rows: 2)
+valid_submission.entries.find_or_create_by!(submission_file: submission_file, aasm_state: "validated", data: { key: "value" }, source: { sheet: "InvoicesReceived", row: 1 })
+valid_submission.entries.find_or_create_by!(submission_file: submission_file, aasm_state: "validated", data: { key: "another value" }, source: { sheet: "InvoicesReceived", row: 2 })
+
+
+framework_invalid = Framework.find_or_create_by!(short_name: 'cboard12', name: 'Cheese Board 12')
+Agreement.find_or_create_by!(supplier: supplier, framework: framework_invalid)
+task_in_review_with_errors = Task.find_or_create_by!(
+  status: :in_progress,
+  due_on: due_date,
+  framework: framework_invalid,
+  supplier: supplier,
+  period_month: Date.today.month,
+  period_year: Date.today.year,
+  description: 'In review task (invalid submission)'
+)
+invalid_submission = supplier.submissions.find_or_create_by!(framework: framework_invalid, task: task_in_review_with_errors, aasm_state: "in_review")
+submission_file = invalid_submission.files.find_or_create_by!(rows: 2)
+invalid_submission.entries.find_or_create_by!(submission_file: submission_file, aasm_state: "validated", data: { key: "value" }, source: { sheet: "InvoicesReceived", row: 1 })
+invalid_submission.entries.find_or_create_by!(
+  submission_file: submission_file,
+  aasm_state: "errored",
+  data: { key: "" },
+  source: { sheet: "InvoicesReceived", row: 2 },
+  validation_errors: { location: { row: 2, column: 1 }, message: 'Required value error' }
```

Comment:
I believe `validation_errors` will either be empty or contain an array.

Reply:
I used the following as prior art:

https://github.com/dxw/DataSubmissionServiceAPI/blob/8473d1afd0f6dbbb3d7270639046c6f62e5c45f6/spec/requests/v1/files_spec.rb#L132-L138

Does that mean this spec is testing validations in the wrong format too?

Reply:
From memory this endpoint was added before we were clear what the `validation_errors` response would look like, so the tests probably should be updated with the correct format.